cli/template/extras/src/env/with-prisma.mjs

```diff
@@ -0,0 +1,77 @@
+// @ts-check
+import { z } from "zod";
+
+/**
+ * Specify your server-side environment variables schema here.
+ * This way you can ensure the app isn't built with invalid env vars.
+ */
+export const server = z.object({
+  DATABASE_URL: z.string().url(),
+  NODE_ENV: z.enum(["development", "test", "production"]),
+});
+
+/**
+ * Specify your client-side environment variables schema here.
+ * This way you can ensure the app isn't built with invalid env vars.
+ * To expose them to the client, prefix them with `NEXT_PUBLIC_`.
+ */
+export const client = z.object({
+  // NEXT_PUBLIC_CLIENTVAR: z.string(),
+});
+
+/**
+ * You can't destruct `process.env` as a regular object in the Next.js
+ * edge runtimes (e.g. middlewares) or client-side so we need to destruct manually.
+ * @type {Record<keyof z.infer<typeof server> | keyof z.infer<typeof client>, string | undefined>}
+ */
+const processEnv = {
+  DATABASE_URL: process.env.DATABASE_URL,
+  NODE_ENV: process.env.NODE_ENV,
+  // NEXT_PUBLIC_CLIENTVAR: process.env.NEXT_PUBLIC_CLIENTVAR,
+};
+
+// Don't touch the part below
+// --------------------------
+
+const formatErrors = (
+  /** @type {z.ZodFormattedError<Map<string,string>,string>} */
+  errors,
+) =>
+  Object.entries(errors)
+    .map(([name, value]) => {
+      if (value && "_errors" in value)
+        return `${name}: ${value._errors.join(", ")}\n`;
+    })
+    .filter(Boolean);
+
+const isServer = typeof window === "undefined";
+
+const merged = server.merge(client);
+const parsed = isServer
+  ? merged.safeParse(processEnv) // on server we can validate all env vars
+  : client.safeParse(processEnv); // on client we can only validate the ones that are exposed
+
+if (parsed.success === false) {
+  console.error(
```

Comment:
imo, we should "streamline" this (below we are using "new Error", and here we are using "console.error"..

Reply:
Actually we do throw here, we just log a nice formatted message before.

![CleanShot 2023-01-20 at 11 25 33](https://user-images.githubusercontent.com/51714798/213673033-2d74e04a-a59e-4b73-8b31-7368429b143d.png)

Reply:
I may have missed this, but perhaps also there a nice formatted message, so the redbox from Next.js is going to tell us straight which are wrong, and we do not need to go into console?